### PR TITLE
Terror Spider Update (pink terrors, spiderling AI, green spiders, etc)

### PR DESCRIPTION
**Pull request by Kyep**
Fake spiderlings replaced with improved spiderling AI
- Spiderlings now try to move to safer areas (areas with more spiders) and away from dangerous areas (areas with more crew). Given the choice between safer areas, they will prefer to hang out with their mother (the queen/green who laid their egg).
- Fake/decoy spiderlings, previously used a lot to give real spiderlings a chance to escape to safety after hatching, have been largely removed.
- This turns spiderling-hunting into a game of skill (are you faster than the spiderling? can you surprise them?) rather than a game of chance (guess which of these 10 identical spiderlings is real).
- In cases where fake spiderlings are deliberately spawned for atmosphere (e.g: Queen's fake spiderlings ability), the fake spiderlings will spawn in greater numbers (25, instead of 15), last 3x as long, and unlike normal spiderlings now, won't attempt to evade threats or hang around the hive. These fake spiderlings are now purely for spooky atmosphere, and will no longer be used as a substitute for threat evasion AI.
- A bug that caused spiderlings to sometimes get permanently stuck has been fixed. They were attempting to move to a vent they could not access (due to a window or similar blocking them). Now, when this happens, they'll give up after a few seconds, and try something else instead.
- Spiderlings will also ventcrawl more often, since there are less of them around, and they need to be smarter about movement. They'll also try to ventcrawl soon after a nearby spiderling dies, so if you walk into a room full of them and kill one, expect the others to try to run away.

New Tier 3 Spider (miniboss): The Princess of Terror
- Starts out pink, fluffy, and cute. Weak in combat against anyone with a weapon. Will provoke "aww" instead of "run for the hills!" - at least until you see it in action. Then it will become nightmare fuel.
- Uses its harmless appearance and ventcrawl ability to move around the station, seeking corpses or easy kills, like a green spider.
- As it consumes more humanoid blood, it changes color, becoming more blood red/purple.
- After consuming the blood of three humanoids, it is strong enough to evolve itself into a queen, revealing its true colors and turning the dark purple used by queens.
- Can be killed by one full clip from an egun, but is generally impervious to civilian-grade weapons, since it has queen-level armor.
- Sprites from FoS:
pink-spider-combined

Green Spider Updates
- Increased HP from 120 to 150. For comparison, reds have 200 HP. This is needed because green infestations were frequently getting slaughtered before they could even lay a single egg.
- Green egg-laying ability now allows the player to choose which type of egg they want to lay. This is an attempt to add more fun and strategic value to playing them.
- Green egg-laying now allows brown/black spider eggs to be laid, but there can only be two of each at any one time. Attempts to create more, even by other green spiders, won't work. The availability of up to 2 browns makes it harder to completely shut down terror spider infestations by welding all vents, and the availability of up to 2 blacks gives normal non-armored crew something to fear.
- Overall, this should make the green spider infestation type less of a joke, and people less reluctant to play greens in general.

Mother Spider Updates
- Mother spiders still spawn 30 spiderlings on death, but now instead of only 10 being real (on average), 15 are real (on average).
- Spiderlings spawned from the death of a mother spider are now exclusively reds, grays, or greens. It is no longer possible mothers to birth black spiders directly (though the greens they create can go on to lay black spider eggs, if fed, and within the above limits).
- New mother spider sprites from FoS:
mother2png

Spider webs now vary slightly based on the type of spider that spins them:
- red: "strong" web has 30 health
- gray: "stealthy" web is faster to spin, and harder to see
- green: "slimy" web causes temporary vision blur
- black: "dark" web causes poisoning
- white: "sickly" web causes egg infection
- brown: no spin web ability, since they should be constantly looking for vents to break open
- purple & prince: "thick" web has 40 health and blocks vision
- princess & queen: web causes hallucination and toxin damage
- mother: no spin web ability, since their job is to die near a vent, not to make a nest

Misc changes
- **All terror spiders now get offered up for ghost control 3 seconds after they're created, rather than the 30 seconds it previously took.** This should reduce "oh cool a spider - oh wait its already dead" syndrome.
- **All terror spiders now default to HARM intent**
- The terror spiders initially spawned by the event can no longer spawn at vents right in front of crew. Nor can they spawn on top of welded vents. Finally, they spawn fully grown, rather than as spiderlings that take a few seconds to grow. This means that infestation events will no longer get aborted before they start because the spiders spawned as spiderlings in the middle of medbay.
- Refactored some vars that were 1/0 to TRUE/FALSE, and consolidated some AI procs

:cl: Kyep
add: Added pink (princess) T3 terror spider, better terror spiderling AI.
tweak: Tweaked green and mother terror spiders. All terror spider types now have their own variants of the standard spider web.
/:cl:

### CODE REVIEW

**Comment by Landerlow** [comment]
YES!

**Comment by Pennwick** [comment]
I like all of this except white spider webs causing egg infection. It just seems too easy to hop from vent to vent leaving webs around blind doorways and getting free infections. Especially since it wouldn't be immediately obvious that a web would somehow infect you internally with spider eggs.

**Comment by Kyep** [comment]
White spider webs already cause infection, and have for a long time. Further, like being bitten by a white spider, you get a bunch of messages in chat after this happens. 

This PR doesn't give white spider webs any new abilities. What it does do is refactor the way special web abilities work to make them more OOP, and to distribute them more evenly. Now, each spider's web has its own special perk, rather than only some spiders (like the white) having them.

**Comment by Shadeykins** [comment]
Do the other webs still block vision? If not that's a pretty severe debuff to ambushing tactics.

**Comment by TDSSS** [comment]
You probably want to add the web changes to the changelog

**Comment by Kyep** [comment]
@Shadeykins @TDSSS  Only the webs that are listed above block vision.

**Comment by Shadeykins** [comment]
Not sure how I feel about that, webs are only really useful for their ability to block vision--most people don't walk into them and laser them from a distance. Out of the dozen or so times I've spooded, I've only ever had four people walk into webs.

**Comment by Kyep** [comment]
@Shadeykins 
I think existing before-this-PR webs are already useful for other purposes, beyond blocking vision. Examples would include area denial, cover (absorbing shots), just roleplaying a spider, etc.
The 'blocks vision' web may be the strongest web that exists currently, but that's intentional. Purples/princes need a very strong defensive web as princes spawn alone and purples' entire job is to protect the queen. So it makes sense that if any spider would have the best web for setting up ambushes, it is these two.
That said, part of this is update is going to be trying to make the other webs, from other spiders, more useful.
For example, under this update, the red spider's web takes more hits to destroy, and the gray's web is partially transparent, rendering it harder to see. Both of those web types make it harder to just laser the web from a distance, and more likely that the web catches someone.
You might have been lucky to have had only 4 people walk into webs. I see it happen quite regularly.

**Comment by Shadeykins** [comment]
I'll have to see how it plays out I guess, I just find the vision blocking ability incredibly useful since it allows for some tactical maneuvering when you would otherwise just be ded.

**Comment by Nicaragua1** [comment]
This improves and adds a lot of content to terror spiders. I love it!.  On the other hand i think terror spiders threat to the station are ok on its current state. What i want to say is that i dont want the terror spiders to be a xeno threat level. Because for that level of threat we already have xenos! So something in between is nice

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/actions.dm [hunk 1]
```diff
@@ -186,6 +188,9 @@
 		return
 	..()
 
+/obj/structure/spider/terrorweb/proc/web_special_ability(var/mob/living/carbon/C)
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/actions.dm]: No var/ please

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/black.dm [hunk 2]
```diff
@@ -42,4 +43,17 @@
...
-		visible_message("<span class='notice'>[src] jumps away from [L]!</span>")
\ No newline at end of file
+		visible_message("<span class='notice'>[src] jumps away from [L]!</span>")
+
+
+/obj/structure/spider/terrorweb/black
+	name = "sinister web"
+	desc = "This web has beads of a dark fluid on its strands."
+
+/obj/structure/spider/terrorweb/black/web_special_ability(var/mob/living/carbon/C)
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/black.dm]: Implicit var

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/princess.dm [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+		desc = "An enormous spider. It looks strangely cute and fluffy, with soft pink fur covering most of its body."
+	else if(fed == 1)
+		icon_state = "terror_princess2"
+		icon_living = "terror_princess2"
+		icon_dead = "terror_princess_dead2"
+		desc = "An enormous spider. It used to look strangely cute and fluffy, but now the effect is spoiled by parts of its fur, which have turned an ominous blood red in color."
+	else
+		icon_state = "terror_princess3"
+		icon_living = "terror_princess3"
+		icon_dead = "terror_princess3_dead"
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/princess.dm]: This is a strange difference in naming convention from the above two

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/actions.dm (hunk 1 above)
Comment (Kyep) [reply code/modules/mob/living/simple_animal/hostile/terror_spiders/actions.dm]: All similar var/s removed.

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/black.dm (hunk 2 above)
Comment (Kyep) [reply code/modules/mob/living/simple_animal/hostile/terror_spiders/black.dm]: All similar var/s removed.

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/princess.dm (hunk 3 above)
Comment (Kyep) [reply code/modules/mob/living/simple_animal/hostile/terror_spiders/princess.dm]: Fixed.

**Pull request merged**